clients/include/testing_matmul.hpp

```diff
@@ -478,13 +478,21 @@ void testing_matmul(const Arguments& arg)
         {
             size_bias[i] = 0;
         }
-        auto biasSize
-            = size_bias[i]
-              * ((arg.d_type != arg.scale_type && arg.bias_type == arg.scale_type) ? sizeof(Talpha)
-                                                                                   : sizeof(To));
+        auto sizeOfBiasType
+            = (arg.d_type != arg.scale_type && arg.bias_type == arg.scale_type) ? sizeof(Talpha)
+                                                                                : sizeof(To);
+        // NB: for in/out = 8bit float and bias type is unset: use Talpha (f32)
+        // TODO: 255 should be an enum of INVALID
+        if((arg.a_type == HIP_R_8F_E4M3_FNUZ || arg.a_type == HIP_R_8F_E5M2_FNUZ)
+            && (arg.b_type == HIP_R_8F_E4M3_FNUZ || arg.b_type == HIP_R_8F_E5M2_FNUZ)
+            && (arg.d_type == HIP_R_8F_E4M3_FNUZ || arg.d_type == HIP_R_8F_E5M2_FNUZ)
+            && (arg.bias_type == 255))
+            sizeOfBiasType = sizeof(Talpha);
+
+        auto totalBiasSize = size_bias[i] * sizeOfBiasType;
         int64_t sizeC = h_beta[i] == 0 ? 0 : size_C[i] * sizeof(To);
```

Comment:
This is meant to align our current ContractionProblem: when bias-type is unset for F8/B8 in/out, we treat it as f32 (Scale type, Talpha). If we don't put this here, then ./hipblaslt-bench makes assumption that is inconsistent with ContractionProblem.